compressors/fano_coder.py

```diff
@@ -0,0 +1,160 @@
+"""Fano Coding
+Shannon coding is a basic prefix free coding. There is some confusion in literature about Shannon, Fano and
+Shannon-Fano codes, e.g. see Naming section of https://en.wikipedia.org/wiki/Shannon–Fano_coding.
+
+This document uses Fano coding as described in the wiki article above.
+"""
+from typing import Any, Tuple
+from utils.bitarray_utils import BitArray
+from utils.test_utils import get_random_data_block, try_lossless_compression
+from compressors.prefix_free_compressors import PrefixFreeTree, PrefixFreeEncoder, PrefixFreeDecoder, BinaryNode
+from core.prob_dist import ProbabilityDist
+
+
+class FanoTree(PrefixFreeTree):
+    def __init__(self, prob_dist):
+        super().__init__(prob_dist)
+        # sort the probability distribution
+        self.sorted_prob_dist = ProbabilityDist.get_sorted_prob_dist(prob_dist.prob_dict)
+        # initialize root node of Fano Tree
+        self.root_node = BinaryNode(id=None)
+
+        # build tree returns the root node
+        super().__init__(root_node=self.build_fano_tree(self.root_node, self.sorted_prob_dist))
+
+
+    @staticmethod
+    def criterion(dict_item):
```

Comment:
I think this should be a local function defined inside `split_prob_dist_into_two` since it is not used anywhere else

Reply:
+1 yes, discussed the same with @tpulkit last week.